make/buf/scripts/release.bash

```diff
@@ -11,137 +11,42 @@ fail() {
+trap "rm -rf $WORKSPACE_DIR" EXIT
+
+test -f "${GOBIN}/${RELEASE_GO_VERSION}" || go install "golang.org/dl/${RELEASE_GO_VERSION}"@latest
+"${GOBIN}/${RELEASE_GO_VERSION}" download 2>/dev/null
 
-for os in Darwin Linux Windows; do
-  for arch in x86_64 arm64; do
-    # our goal is to have the binaries be suffixed with $(uname -s)-$(uname -m)
-    # on mac, this is arm64, on linux, this is aarch64, for historical reasons
-    # this is a hacky way to not have to rewrite this loop (and others below)
-    if [ "${os}" == "Linux" ] && [ "${arch}" == "arm64" ]; then
-      arch="aarch64"
-    fi
-    extension=""
-    if [ "${os}" == "Windows" ]; then
-      extension=".exe"
-    fi
-    dir="${os}/${arch}/${BASE_NAME}"
-    mkdir -p "${dir}/bin"
-    for binary in \
-      buf \
-      protoc-gen-buf-breaking \
-      protoc-gen-buf-lint; do
-      CGO_ENABLED=0 GOOS=$(goos "${os}") GOARCH=$(goarch "${arch}") \
-        go build -a -ldflags "-s -w" -trimpath -o "${dir}/bin/${binary}${extension}" "${DIR}/cmd/${binary}/main.go"
-      cp "${dir}/bin/${binary}${extension}" "${binary}-${os}-${arch}${extension}"
-    done
-  done
-done
+ln -s "${GOBIN}/${RELEASE_GO_VERSION}" "${RELEASE_DIR}/go"
 
-for os in Darwin Linux; do
-  for arch in x86_64 arm64; do
-    if [ "${os}" == "Linux" ] && [ "${arch}" == "arm64" ]; then
-      arch="aarch64"
-    fi
-    dir="${os}/${arch}/${BASE_NAME}"
-    mkdir -p "${dir}/etc/bash_completion.d"
-    mkdir -p "${dir}/share/fish/vendor_completions.d"
-    mkdir -p "${dir}/share/zsh/site-functions"
-    #mkdir -p "${dir}/share/man/man1"
-    "$(uname -s)/$(uname -m)/${BASE_NAME}/bin/buf" bash-completion > "${dir}/etc/bash_completion.d/buf"
-    "$(uname -s)/$(uname -m)/${BASE_NAME}/bin/buf" fish-completion > "${dir}/share/fish/vendor_completions.d/buf.fish"
-    "$(uname -s)/$(uname -m)/${BASE_NAME}/bin/buf" zsh-completion > "${dir}/share/zsh/site-functions/_buf"
-    #"$(uname -s)/$(uname -m)/${1}/bin/buf" manpages "${dir}/share/man/man1"
-    cp -R "${DIR}/LICENSE" "${dir}/LICENSE"
-  done
-done
+echo "${RELEASE_MINISIGN_PRIVATE_KEY}" >"${WORKSPACE_DIR}/minisignsecret"
+echo "${RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD}" >"${WORKSPACE_DIR}/minisignpass"
 
-for os in Darwin Linux; do
-  for arch in x86_64 arm64; do
-    if [ "${os}" == "Linux" ] && [ "${arch}" == "arm64" ]; then
-      arch="aarch64"
-    fi
-    dir="${os}/${arch}/${BASE_NAME}"
-    tar_context_dir="$(dirname "${dir}")"
-    tar_dir="${BASE_NAME}"
-    tarball="${BASE_NAME}-${os}-${arch}.tar.gz"
-    tar -C "${tar_context_dir}" -cvzf "${tarball}" "${tar_dir}"
-  done
-done
+mkdir -p "${WORKSPACE_DIR}/etc/bash_completion.d" \
+  "${WORKSPACE_DIR}/share/fish/vendor_completions.d" \
+  "${WORKSPACE_DIR}/share/zsh/site-functions"
 
-for file in $(find . -maxdepth 1 -type f | sed 's/^\.\///' | sort | uniq); do
-  sha256 "${file}" >> sha256.txt
-done
-sha256 -c sha256.txt
+"${GOBIN}/buf" bash-completion >"${WORKSPACE_DIR}/etc/bash_completion.d/buf"
```

Comment:
This should not use the project `buf` installation, it should use the release `buf` build